shell/platform/android/io/flutter/plugin/common/MethodChannel.java

```diff
@@ -152,6 +156,54 @@ public void setMethodCallHandler(final MethodCallHandler handler) {
         void notImplemented();
     }
 
+    /**
+     * An exception that {@link MethodCallHandler} implementations may throw in
+     * {@link MethodCallHandler#onMethodCall(MethodCall, Result)} that will be caught by the
+     * channel implementation and sent back to Flutter as a {@code PlatformException}. Metadata
+     * in this exception class will be translated to corresponding fields in the Dart exception.
+     *
+     * <p>Other uncaught exceptions thrown by {@code MethodCallHandler} implementations will also
+     * be sent back to Flutter as a {@code PlatformException}, but they will have a generic
+     * "error" code and no error details.</p>
+     */
+    public static final class ResultException extends RuntimeException {
```

Comment:
The reason is better control flow. I no longer have to pass around result just to be able to set the error and/or I don't have to carry around boolean return values to indicate when the main onMethodCall fn should return w/o doing anything else.

is there a use case where you want to set result.error() and keep going? I doubt it.